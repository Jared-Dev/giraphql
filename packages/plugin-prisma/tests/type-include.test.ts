import { execute } from 'graphql';
import { gql } from 'graphql-tag';
import { prisma } from './example/builder';
import schema from './example/schema';

let queries: unknown[] = [];
prisma.$use((params, next) => {
  queries.push(params);

  return next(params);
});

describe('prisma', () => {
  afterEach(() => {
    queries = [];
  });

  afterAll(async () => {
    await prisma.$disconnect();
  });

  it('loads type-level includes directly, indirectly and merges with relations', async () => {
    const query = gql`
      query {
        me {
          commentsConnection(first: 1) {
            edges {
              node {
                authorBio
              }
            }
          }
          twoComments: commentsConnection(first: 2) {
            edges {
              node {
                authorBio
                author {
                  id
                }
              }
            }
          }
          postsConnection(first: 1) {
            edges {
              node {
                commentAuthorIds
              }
            }
          }
        }
      }
    `;

    const result = await execute({
      schema,
      document: query,
      contextValue: { user: { id: 1 } },
    });

    expect(result).toMatchSnapshot();

    expect(queries).toMatchInlineSnapshot(`
      [
        {
          "action": "findUnique",
          "args": {
            "include": {
              "comments": {
                "include": {
                  "author": {
                    "include": {
                      "profile": true,
                    },
                  },
                },
                "skip": 0,
                "take": 2,
              },
              "posts": {
                "include": {
                  "comments": {
                    "include": {
                      "author": true,
                    },
                    "take": 3,
                  },
                },
                "orderBy": {
                  "createdAt": "desc",
                },
                "skip": 0,
                "take": 2,
              },
            },
            "where": {
              "id": 1,
            },
          },
          "dataPath": [],
          "model": "User",
          "runInTransaction": false,
        },
        {
          "action": "findUniqueOrThrow",
          "args": {
            "include": {
              "comments": {
                "include": {
                  "author": {
                    "include": {
                      "profile": true,
                    },
                  },
                },
                "skip": 0,
                "take": 3,
              },
            },
            "where": {
              "id": 1,
            },
          },
          "dataPath": [],
          "model": "User",
          "runInTransaction": false,
        },
      ]
    `);
  });
});
