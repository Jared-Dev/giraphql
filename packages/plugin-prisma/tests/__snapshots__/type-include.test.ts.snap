// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`prisma > loads type-level includes directly, indirectly and merges with relations 1`] = `
{
  "data": {
    "me": {
      "commentsConnection": {
        "edges": [
          {
            "node": {
              "authorBio": "Debitis perspiciatis unde sunt.",
            },
          },
        ],
      },
      "postsConnection": {
        "edges": [
          {
            "node": {
              "commentAuthorIds": [
                "1",
                "2",
                "3",
              ],
            },
          },
        ],
      },
      "twoComments": {
        "edges": [
          {
            "node": {
              "author": {
                "id": "VXNlcjox",
              },
              "authorBio": "Debitis perspiciatis unde sunt.",
            },
          },
          {
            "node": {
              "author": {
                "id": "VXNlcjox",
              },
              "authorBio": "Debitis perspiciatis unde sunt.",
            },
          },
        ],
      },
    },
  },
}
`;
